beacon-chain/p2p/fork_watcher.go

```diff
@@ -0,0 +1,30 @@
+package p2p
+
+import (
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/prysmaticlabs/prysm/shared/slotutil"
+)
+
+// A background routine which listens for new and upcoming forks and
+// updates the node's discovery service to reflect any new fork version
+// changes.
+func (s *Service) forkWatcher() {
+	slotTicker := slotutil.NewSlotTicker(s.genesisTime, params.BeaconConfig().SecondsPerSlot)
+	for {
+		select {
+		case currSlot := <-slotTicker.C():
+			currEpoch := helpers.SlotToEpoch(currSlot)
+			if currEpoch == params.BeaconConfig().AltairForkEpoch {
```

Comment:
Concern here: what if a node is started after the altair fork epoch? It will miss this line of code because this depends on an actual clock ticker. Is this a problem?

Reply:
Not really, there is a path at startup which handles any fork changes( via epoch). So a node starting up later will not miss it.